README.md

```
@@ -63,6 +63,30 @@ This project keeps type safety in plain JavaScript by relying on JSDoc plus the
+### Example Type Error
+
+Type checking often catches null/undefined paths before they hit runtime. For example, accessing a track before verifying it exists will fail `npm run check:tsc`:
+
+```js
+/** @param {{ tracks?: import('./types.js').NormalizedTrack[] }} playlist */
+function describeFirstTrack(playlist) {
+  return playlist.tracks[0].title.toLowerCase();
+}
+```
+
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)